test/Graphics/Rect.cpp

```diff
@@ -69,31 +69,18 @@ TEST_CASE("sf::Rect class template - [graphics]")
 
     SUBCASE("Intersection")
     {
-        SUBCASE("intersects(Rect)")
-        {
-            sf::IntRect rectangle({0, 0}, {10, 10});
-            sf::IntRect intersectingRectangle({5, 5}, {10, 10});
-            sf::IntRect nonIntersectingRectangle({-5, -5}, {5, 5});
-
-            CHECK(rectangle.intersects(intersectingRectangle) == true);
-            CHECK(rectangle.intersects(nonIntersectingRectangle) == false);
-        }
-
-        SUBCASE("intersects(Rect, Rect)")
-        {
-            sf::IntRect rectangle({0, 0}, {10, 10});
-            sf::IntRect intersectingRectangle({5, 5}, {10, 10});
-            sf::IntRect nonIntersectingRectangle({-5, -5}, {5, 5});
-            sf::IntRect intersectionResult;
-
-            CHECK(rectangle.intersects(intersectingRectangle, intersectionResult) == true);
-            CHECK(intersectionResult.top == 5);
-            CHECK(intersectionResult.left == 5);
-            CHECK(intersectionResult.width == 5);
-            CHECK(intersectionResult.height == 5);
-
-            CHECK(rectangle.intersects(nonIntersectingRectangle, intersectionResult) == false);
-        }
+        const sf::IntRect rectangle({0, 0}, {10, 10});
+        const sf::IntRect intersectingRectangle({5, 5}, {10, 10});
+
+        const auto intersectionResult = rectangle.intersects(intersectingRectangle);
+        CHECK(intersectionResult.has_value());
```

Comment:
```suggestion
        REQUIRE(intersectionResult.has_value());
```

Otherwise the `->` accesses will be UB.